src/core/qgis.h

```diff
@@ -6256,6 +6258,34 @@ CORE_EXPORT bool qgsVariantEqual( const QVariant &lhs, const QVariant &rhs );
  */
 CORE_EXPORT bool qgsVariantGreaterThan( const QVariant &lhs, const QVariant &rhs );
 
+/**
+ * Returns whether the \a scale is equal to or greater than the \a minScale,
+ * taking non-round numbers into account.
+ *
+ * \param scale The current scale to be compared.
+ * \param minScale The minimum map scale (i.e. most "zoomed out" scale) at
+ * which features, labels or diagrams will be visible. The scale value
+ * indicates the scale denominator, e.g. 1000.0 for a 1:1000 map.
+ * \see qgsLessThanMaximumScale()
+ *
+ * \since QGIS 3.40
+ */
+CORE_EXPORT bool qgsEqualToOrGreaterThanMinimumScale( const double scale, const double minScale );
```

Comment:
Could this be moved to static function in a "QgsScaleUtils" class instead?

Reply:
Yes, done in https://github.com/qgis/QGIS/pull/58968/commits/2b4debadb0f08507411bb5c147c1c6348ac51aea